CustomAlerts.php

```diff
@@ -183,6 +189,43 @@ public function getSiteIdsHavingAlerts()
         return array_values(array_unique($siteIdsHavingAlerts));
     }
 
+    /**
+     * If the task is for CustomAlerts, save the name of the task in the option so that we know what the currently
+     * running task is.
+     *
+     * @param Task $task
+     * @return void
+     */
+    public function startingScheduledTask(Task $task): void
+    {
+        if (strpos($taskName = $task->getName(), 'Piwik\Plugins\CustomAlerts\Tasks') === false) {
+            return;
+        }
+
+        /** @var \Piwik\Scheduler\Timetable $timetable */
+        $timetable = StaticContainer::getContainer()->get('Piwik\Scheduler\Timetable');
+        // Look up the retry count so that we know whether this is a retry or not
+        $retryCount = $timetable->getRetryCount($taskName);
+        $optionJson = json_encode(['taskName' => $taskName, 'retryCount' => $retryCount]);
+
+        Option::set(self::CUSTOM_ALERTS_CURRENT_SCHEDULED_TASK_OPTION, $optionJson);
```

Comment:
Is there any reason for storing this in an additional option? The retry count is already stored in the timetable.
A simpler approach could be to only store the name of the current running task in an static property and directly fetch the retry count from timetable where it is needed.

Reply:
I guess I could try using a status property to hold the task name. I was just trying to come up with a way to track it since there isn't currently a built-in method in the Scheduler class and I'm trying to avoid increasing the minimum required version. I figured that I would look up the retry count while I had the task name readily available, but I can try getting rid of the option.

Reply:
Thank you for the suggestion @sgiehl . Switching to a static property appears to do the job fine :+1: